### PR TITLE
feat: add clang-format config

### PR DESCRIPTION
**Pull request by black-desk**

### CODE REVIEW

**Comment by black-desk** [comment]
https://github.com/linuxdeepin/dtksystemsettings/pull/128

File: qt/qt.editorconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[*.{cpp,h,cc,hpp}]
+
+indent_style = space
+indent_size = 4
+tab_width = 8
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
```
Comment (black-desk) [inline qt/qt.editorconfig:8]: @zsien 给你加上了

Comment (zsien) [reply qt/qt.editorconfig:8]: 等等，tab_width = 8？

Comment (black-desk) [reply qt/qt.editorconfig:8]: 往上看 缩进不使用tab, tab的默认值就是8

Comment (zsien) [reply qt/qt.editorconfig:8]: 好的呢

**Comment by black-desk** [comment]
@Groveer

**Comment by Groveer** [comment]
@Iceyer

**Comment by black-desk** [comment]
> 1. Should we remove `SortIncludes: true`?

I keep thinking about the include issue these days. Now I have an opinion that we should not sort includes unless we use something like include-what-you-use as a linter to keep includes clean.

> 2. Should keep space after `{` like this or not:
>    `cacheAbsoluteFilePath({ "runtime", "files" });`

I like this.

> 3. `ContinuationIndentWidth: 8` --> `ContinuationIndentWidth: 4` is better?

This is come from qt.

I check the blame it seems like they update indent size from 8 to 4 and forget to update this setting. I will change this later.

refer: qt5 git repo commit 5872e22f3

> 4. Ungly line break should be fixed:
> 
> ```diff
> -        if (infoFile.write(QJsonDocument::fromVariant(toVariant<package::Info>(&info)).toJson()) < 0) {
> +        if (infoFile.write(QJsonDocument::fromVariant(toVariant<package::Info>(&info)).toJson())
> +            < 0) {
>              return NewError(infoFile.error(), infoFile.errorString());
>          }
> ```

Let me have a try.

**Comment by black-desk** [comment]
> 4\. Ungly line break should be fixed:
> 
> ```diff
> -        if (infoFile.write(QJsonDocument::fromVariant(toVariant<package::Info>(&info)).toJson()) < 0) {
> +        if (infoFile.write(QJsonDocument::fromVariant(toVariant<package::Info>(&info)).toJson())
> +            < 0) {
>              return NewError(infoFile.error(), infoFile.errorString());
>          }
> ```

Unfortunately, clang-format do not have a penalty for break before operator.

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#penaltybreakassignment

**Comment by black-desk** [comment]
@Groveer ping

**Review by Groveer — APPROVED** [review]

**Pull request merged**